### PR TITLE
LPS-143637 Avoid using UserPermissionUtil, BrowserSnifferUtil, DirectRequestDispatcherFactoryUtil directly in modules

### PR DESCRIPTION
**Pull request by julschong**
Hi Tina, Shuyang

Main Task: [LPS-143403](https://issues.liferay.com/browse/LPS-143403)
Sub Task 10: [LPS-143637](https://issues.liferay.com/browse/LPS-143637)

Note:
- The last commit is to patch additional RolePermissionUtil that got merged in together from #377 now causing SF error on AccountRoleServiceImpl
- BroswerSnifferUtil has 1 usage in a static context, cannot be fully removed yet

Thank you!

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test

**Comment by julschong** [comment]
@kyle-miho
Hi Kyle,
Could you please help me scan the test results?
Thank you!

**Comment by kyle-miho** [comment]
@julschong failures seem unrelated

**Comment by julschong** [comment]
@tinatian
Hi Tina, could you please help review this PR?
Thank you!

File: modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/frontend/taglib/servlet/taglib/AccountUserAccountEntriesScreenNavigationEntry.java [hunk 1]
```diff
@@ -103,4 +103,7 @@ public void render(
 	@Reference
 	protected JSPRenderer jspRenderer;
 
+	@Reference
+	protected UserPermission userPermission;
```
Comment (tinatian) [inline modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/frontend/taglib/servlet/taglib/AccountUserAccountEntriesScreenNavigationEntry.java:107]: @julschong Why this is protected? No one extends this class.

**Comment by tinatian** [comment]
@julschong You removed 4 ***Util usages in modules, but you only add 2 of them to SF properties, is there any reason?

File: modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/frontend/taglib/servlet/taglib/AccountUserAccountEntriesScreenNavigationEntry.java (hunk 1 above)
Comment (julschong) [reply modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/frontend/taglib/servlet/taglib/AccountUserAccountEntriesScreenNavigationEntry.java:107]: @tinatian Hi Tina, I think I just followed the JSPRenderer service reference already there. I'll make this private and resubmit it

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test

**Comment by julschong** [comment]
> @julschong You removed 4 ***Util usages in modules, but you only add 2 of them to SF properties, is there any reason?

@tinatian 
- BrowserSnifferUtil has a static context that I cannot change to use service reference: 
    com.liferay.dynamic.data.mapping.internal.util.DDMFormFieldFreeMarkerRendererUtil
- RolePermissionUtil [LPS-143636](https://issues.liferay.com/browse/LPS-143636) was merged in.
    But 1 extra usage of this util was added at the same time with that PR. 
    Alan reported [here](https://github.com/ling-alan-huang/liferay-portal/pull/710#issuecomment-1057109948) and I checked it again.

**Comment by ling-alan-huang** [comment]
SF failures is unrelated, fix is on the way https://github.com/brianchandotcom/liferay-portal/pull/114174

**Comment by tinatian** [comment]
ci:test:sf

**Comment by ling-alan-huang** [comment]
ci:test:sf

**Comment by tinatian** [comment]
@shuyangzhou This pull is good to go.

Tina.

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/114299

**Comment by shuyangzhou** [comment]
ci:close

**Pull request closed without merging**